mkosi/machine.py

```diff
@@ -171,7 +171,10 @@ def setUpClass(cls) -> None:
         if os.getuid() != 0:
             raise unittest.SkipTest("Must be invoked as root.")
 
-        cls.machine = Machine(cls.args)
+        try:
+            cls.machine = Machine(cls.args)
+        except MkosiException as exception:
+            raise unittest.SkipTest(str(exception))
```

Comment:
I removed the `if`'s from ci.yml and passed the invalid combinations of distribution and format to Mkosi anyways. __init__.py raises an exception if such combination is invalid, then, we skip.